src/Microsoft.VisualStudio.Services.Agent/JobServer.cs

```diff
@@ -142,12 +153,46 @@ public async Task<BlobIdentifierWithBlocks> UploadLogToBlobstorageService(Stream
             // Since we read this while calculating the hash, the position needs to be reset before we send this 
             blob.Position = 0;
 
-            using(var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
+            using (var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
             {
                 return await blobClient.UploadBlocksForBlobAsync(blobId, blob, default(CancellationToken));
             }
         }
 
+        public async Task<(DedupIdentifier dedupId, ulong length)> UploadToBlobStore(bool verbose, string itemPath, CancellationToken cancellationToken)
+        {
+            var (dedupClient, clientTelemetry) = await DedupManifestArtifactClientFactory.Instance
+                    .CreateDedupClientAsync(verbose, (str) => Trace.Info(str), this._connection, cancellationToken);
+
+            // Create chunks and identifier
+            var chunk = await ChunkerHelper.CreateFromFileAsync(FileSystem.Instance, itemPath, cancellationToken, false);
+            var rootNode = new DedupNode(new []{ chunk});
+            var dedupId = rootNode.GetDedupIdentifier(HashType.Dedup64K);
+
+            // Setup upload session to keep file for at mimimum one day
+            var tracer = DedupManifestArtifactClientFactory.CreateArtifactsTracer(verbose, (str) => Trace.Info(str));
+            var keepUntulRef = new KeepUntilBlobReference(DateTime.UtcNow.AddDays(1));
```

Comment:
Nit: Should be keepUntilRef

Reply:
Also, we're safe to set this keep until ref to just 1 day because we're later going to associate a reference with it on the server, right? That might be a good thing to call out in the comment

Reply:
I added a comment